File: 백준/Bronze/28940. Дневник Гравити Фолз/Дневник Гравити Фолз.c
#include <stdio.h>
int main(void) {
    int w, h, n, a, b, rlt;
    scanf("%d %d", &w, &h);
    scanf("%d %d %d", &n, &a, &b);
    rlt = (w / a) * (h / b);
    if (w < a || h < b)
        printf("%d", -1);
    else
        printf("%d", n % rlt == 0 ? n / rlt : n / rlt + 1);
    return 0;
}

File: 백준/Bronze/28940. Дневник Гравити Фолз/Дневник Гравити Фолз.py
import sys
input=lambda:sys.stdin.readline().rstrip()
w,h=map(int,input().split())
n,a,b=map(int,input().split())
rlt=(w//a)*(h//b)
if w<a or h<b:
     print(-1)
else:
     print(int(n//rlt)+(1 if n%rlt!=0 else 0))